### DISCUSSION TITLE
Autonomous Navigation for Vehicles in Rough Terrain

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#40] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to[ project #209](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/tree/main/projects/Autonomous%20Navigation%20for%20Vehicles%20in%20Rough%20Terrain)

**Comment by raj1652** [comment]
Hi, I am a Mechatronics student at Terna Engineering College and interested in working and collaborating on this project.

**Reply by 2003Vandu** [reply]
i am intrested i am from Automation robotic Branch

**Comment by sskulka** [comment]
Hi Roberto,
I am Shubhankar from Clemson University, SC, USA. I'm facing an issue while working on the project. Is there a way to make the co-simulation compatible with gazebo 11? I am facing an issue running the gazebo9 with my ros noetic due to certain packages not being compatible with it. Is there a solution to this?

Thanks & Regards,
Shubhankar Kulkarni

**Reply by sskulka** [reply]
![image](https://user-images.githubusercontent.com/90679857/144687849-65a8709d-44fb-4c03-990e-37e1219649de.png)

**Comment by lgavshin** [comment]
Hi @sskulka,

Thanks for reaching out to us with the your problem. Unfortunately as of now we support up to Gazebo 10.x for the released versions. We are working on adding the support for Gazebo 11.x.

As of now as a temporary measure try removing/commenting line which is generating the error.  Depending on the environment and use-case it might work in this case. But as of now we haven't tested and we also don't have details about the development environment the code is compiled in, so we can not be sure that it would work.

You should be able to find the file in the GazeboPlugin ".zip" file created after 'packageGazeboPlugin' is executed in MATLAB. You can search for GAZEBO_MAX_SUPPORTED_VERSION to find the file and line to modify.

Let us know if it works for you it might help out others who've signed up for the project.

Please feel free to reach out to me in case you have any more questions regarding the project.

Thanks,
Laxit

**Reply by lgavshin** [reply]
Alternately you can also try connecting to gazebo through ROS framework, this [example](https://www.mathworks.com/help/ros/ug/get-started-with-ros-in-simulink.html) can get you started on how to interact with ROS or this [example](https://www.mathworks.com/help/robotics/ug/pick-and-place-workflow-in-gazebo-using-ros.html) to interact with Gazebo via ROS.

**Reply by sskulka** [reply]
Hi @lgavshin , I was able to get around the issue using the second alternative which you had mentioned (of connecting  to gazebo through ROS framework). I tried the Gazebo max version commenting out workaround but was still facing the issue after which I switched to the ROS framework method.
Thank you!

**Reply by lgavshin** [reply]
Thanks for the confirmation @sskulka.

**Comment by Hrishikesh-H** [comment]
Hi, I am a student at IIT Indore and interested in working and collaborating on this project.

**Reply by lgavshin** [reply]
Hi Hrishikesh,

I am glad to see that you are interested in the project. Please don't hesitate to reach out in case you face any issues.

**Reply by Phani69** [reply]
im also interested

**Comment by Vibha-Bhalerao** [comment]
Hi, I am a B.Tech. Mechanical Engineering student at the National Institute of Technology Trichy, and interested in working on this project. 

Before starting, I want to confirm the correct software stack for my environment:

My current System Setup:
- Windows 11 with WSL2
- MATLAB R2025a (with Robotics System Toolbox + Navigation Toolbox)
- Target: Ubuntu 22.04 LTS in WSL2
- Hardware: 8 GB RAM (Dell Vostro i5)

Planned Stack:
1. Ubuntu 22.04 LTS
2. ROS2 Humble
3. Gazebo Fortress (or Classic 11?)
4. MATLAB-ROS2 bridge

Questions:
1. Is Ubuntu 22.04 LTS the recommended choice for MATLAB R2025a + Gazebo + ROS2 Humble?
2. Should I use Gazebo Fortress or Gazebo Classic 11? Which is better supported 
   with MATLAB R2025a?
3. Is the Ubuntu 22.04 + ROS2 Humble + Gazebo Fortress combination officially 
   tested and stable?
4. Any known issues with WSL2 + this stack regarding 8 GB RAM constraints? If so, is there any way to optimize the solutions while keeping the 8 GB constraint?
5. Is Gazebo co-simulation with Simulink (not just ROS2 bridge) supported with 
   Gazebo Fortress on WSL2?

Inputs on the above queries will help me ensure the most stable, well-supported combination to avoid compatibility issues mid-project.

Any guidance would be appreciated!

**Reply by mohit-Mathworks** [reply]
Hi Vibha, thanks for your interest! Here are responses to your environment related questions.

> 1. Is Ubuntu 22.04 LTS recommended for MATLAB R2025a + ROS2 Humble + Gazebo?

Yes — Ubuntu 22.04 is the officially supported OS for both [ROS2 ](https://docs.ros.org/en/humble/Installation.html) & MATLAB [R2025a](https://www.mathworks.com/content/dam/mathworks/mathworks-dot-com/support/sysreq/files/[system-requirements-release-2025a-linux.pdf](https://www.mathworks.com/content/dam/mathworks/mathworks-dot-com/support/sysreq/files/system-requirements-release-2025a-linux.pdf)). 

>2 & 3.  Should I use Gazebo Fortress or Gazebo Classic 11? Which is better supported with MATLAB R2025a?

[MATLAB/Simulink officially supports Gazebo Classic 11](https://www.mathworks.com/help/ros/ug/install-and-set-up-virtual-machine-for-ros-and-gazebo.html) for ROS/ROS 2 integration and co‑simulation.

> 4. RAM limitations: known issues with WSL2 + this stack on 8 GB RAM?

With 8 GB RAM, here are some practices I recommend:
- Prefer Gazebo Classic 11 (lighter).
- Run Gazebo in headless mode when possible (gzserver without gzclient).
- Avoid running heavy Gazebo scenes + MATLAB simultaneously inside WSL2.

> 5. Is Gazebo co‑simulation with Simulink supported using Gazebo Fortress on WSL2?

Simulink co‑simulation has been officially tested [with Gazebo Classic 11](https://www.mathworks.com/help/ros/ug/install-and-set-up-virtual-machine-for-ros-and-gazebo.html); support for Gazebo Fortress has not been verified. 

Please feel free to reach out to me in case you have any more questions regarding the project.

Thanks,
Mohit
